packages/ui/src/views/repo/repo-tags/components/repo-tags-list.tsx

```diff
@@ -1,76 +1,153 @@
-import { CommitCopyActions, MoreActionsTooltip, SkeletonTable, Table } from '@/components'
-import { formatDate } from '@utils/utils'
-import { TranslationStore } from '@views/repo/repo-list/types'
+import { FC, useMemo } from 'react'
 
-import { RepoTagsStore } from '../types'
+import { Avatar, CommitCopyActions, MoreActionsTooltip, NoData, SkeletonTable, Table, Text } from '@/components'
+import { getInitials, timeAgo } from '@/utils'
+import { BranchSelectorListItem, CommitTagType, RepoTagsStore, TranslationStore } from '@/views'
 
 interface RepoTagsListProps {
   useTranslationStore: () => TranslationStore
-  isLoading: boolean
   onDeleteTag: (tagName: string) => void
   useRepoTagsStore: () => RepoTagsStore
   toCommitDetails?: ({ sha }: { sha: string }) => string
-  openCreateBranchDialog: () => void
+  openCreateBranchDialog: (selectedTagInList: BranchSelectorListItem) => void
+  isLoading: boolean
+  isDirtyList: boolean
+  handleResetFiltersAndPages: () => void
+  openCreateTagDialog: () => void
 }
-export const RepoTagsList: React.FC<RepoTagsListProps> = ({
+
+export const RepoTagsList: FC<RepoTagsListProps> = ({
   useTranslationStore,
-  isLoading,
   onDeleteTag,
   useRepoTagsStore,
   toCommitDetails,
-  openCreateBranchDialog
+  openCreateBranchDialog,
+  isLoading,
+  isDirtyList,
+  handleResetFiltersAndPages,
+  openCreateTagDialog
 }) => {
   const { t } = useTranslationStore()
   const { tags: tagsList } = useRepoTagsStore()
 
+  const getTableActions = useMemo(
```

Comment:
There's no benefit in memoizing this function since it's used within this component. The return of the function will still be new on every render. The overhead of memoization will actually slow things down as it is now.